android-core/src/main/java/com/mparticle/networking/MParticleBaseClientImpl.java

```diff
@@ -103,26 +103,34 @@ protected MPUrl getUrl(Endpoint endpoint, @Nullable String identityPath, boolean
         NetworkOptions networkOptions = mConfigManager.getNetworkOptions();
         DomainMapping domainMapping = networkOptions.getDomain(endpoint);
         String url = NetworkOptionsManager.getDefaultUrl(endpoint);
+        // Default domain to use for URL generation when domain mapping is specified
+        String defaultDomain = url ;
         boolean isDefaultUrl = true;
-        if (domainMapping != null && !MPUtility.isEmpty(domainMapping.getUrl()) && !forceDefaultUrl) {
+        if (domainMapping != null && !MPUtility.isEmpty(domainMapping.getUrl())) {
```

Comment:
if `forceDefaultUrl` is removed from here, we should remove it from the function signature as well

Reply:
As discussed, we will not remove it at this time, as it would require significant changes to the getURL method.